docs/customization/factory.rst

```diff
@@ -0,0 +1,126 @@
+Customizing Factories
+=====================
+
+.. warning::
+
+    In **Sylius** we are already decorating factories from components in Core.
+    Often you will be needing to add your very own methods to them. You need to check before which factory is your resource using.
+
+Why would you customize a Factory?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Differently configured versions of resources may be needed in various scenarios in your application.
+You may need for instance to:
+
+    * create Product with Supplier(which is your own custom entity)
+    * create a disabled Product(for further modifications)
+    * create a ProductReview with predefined description
+
+and many, many more usecases.
+
+How to customize a Factory?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's assume that you would want to have a possibility to create disabled products.
+
+1. Create your own factory class in the ``AppBundle\Factory`` namespace.
+Remember that it has to extend a proper base class. How can you check that?
+
+For the ``ProductFactory`` run:
+
+.. code-block:: bash
+
+    $ php bin/console debug:container sylius.factory.product
+
+As a result you will get the ``Sylius\Component\Product\Factory\ProductFactory`` - this is the class that you need to be extending.
```

Comment:
We will receive a table actually. Something like: `Check class option. It is value is a class that we are interested in`